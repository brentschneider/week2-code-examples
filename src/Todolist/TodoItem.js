import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Todolist.css';

class TodoItem extends Component {
    // Another way of defining propTypes
    // static propTypes = {
    //     text: PropTypes.string.isRequired
    // }

    render() {
        return <li>{this.props.text}</li>;
    }
}

TodoItem.propTypes = {
    text: PropTypes.string.isRequired
};

export default TodoItem;