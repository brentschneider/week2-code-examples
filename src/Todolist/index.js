import React, { Component } from 'react';
import TodoItem from './TodoItem';

export class Todolist extends Component {
    constructor(props) {
        super(props);
        this.state = {
            items: [],
            inputText: ''
        };
    }

    onInputChange = (e) => {
        const inputText = e.target.value;
        this.setState((prevState, props) => {
            return {
                inputText: inputText
            };
        });
    }

    addItem = () => {
        this.setState((prevState, props) => {
            const items = prevState.items;
            const item = prevState.inputText;
            if (item) {
                items.push(item);
            } else {
                alert('You should enter an item! ðŸ˜‰');
            }

            return {
                items: items,
                inputText: ''
            };
        });
    }

    render() {
        const listItems = this.state.items.map((item, idx) => {
            return <TodoItem key={idx} text={item} />;
        });

        return (
            <div>
                <h1>Todo List</h1>
                <div>
                    <input
                        onChange={this.onInputChange}
                        value={this.state.inputText}
                    />
                    <button
                        disabled={this.state.inputText === ''}
                        onClick={this.addItem}
                    >Add Item</button>
                </div>
                <ul>
                    {listItems}
                </ul>
            </div>
        );
    }
}

export default Todolist;
