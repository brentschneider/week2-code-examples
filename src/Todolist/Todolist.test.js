import React from 'react';
import Todolist from './index';
import ReactTestRenderer from 'react-test-renderer';

describe('Adding numbers', () => {
    it('should add two numbers', () => {
        expect(2 + 2).toBe(4);
    });
});

describe('Todolist component', () => {
    it('should render', () => {
        const component = ReactTestRenderer.create(<Todolist />);    
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('change the input', () => {
        const component = ReactTestRenderer.create(<Todolist />);
        component.getInstance().onInputChange({
            target: {
                value: 'new value'
            }
        });
        expect(component.toJSON()).toMatchSnapshot();
    });
});
