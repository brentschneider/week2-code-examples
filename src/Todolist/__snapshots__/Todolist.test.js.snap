// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todolist component change the input 1`] = `
<div>
  <h1>
    Todo List
  </h1>
  <div>
    <input
      onChange={[Function]}
      value="new value"
    />
    <button
      disabled={false}
      onClick={[Function]}
    >
      Add Item
    </button>
  </div>
  <ul />
</div>
`;

exports[`Todolist component should render 1`] = `
<div>
  <h1>
    Todo List
  </h1>
  <div>
    <input
      onChange={[Function]}
      value=""
    />
    <button
      disabled={true}
      onClick={[Function]}
    >
      Add Item
    </button>
  </div>
  <ul />
</div>
`;
